projects/cds/src/merchandising/adapters/strategy/merchandising.strategy.adapter.spec.ts

```diff
@@ -0,0 +1,100 @@
+import {
+  HttpClientTestingModule,
+  HttpTestingController,
+} from '@angular/common/http/testing';
+import { Type } from '@angular/core';
+import { TestBed } from '@angular/core/testing';
+import { CdsEndpointsService } from '../../../services/cds-endpoints.service';
+import { StrategyResult } from '../../model/strategy.result';
+import { MerchandisingStrategyAdapter } from './merchandising.strategy.adapter';
+import createSpy = jasmine.createSpy;
+
+const strategyId = 'test-strategy-id';
+
+const strategyResultMetadata: Map<string, string> = new Map<string, string>();
+strategyResultMetadata.set('test-metadata-field', 'test-metadata-value');
+const productMetadata: Map<string, string> = new Map<string, string>();
+productMetadata.set(
+  'test-product-metadata-field',
+  'test-product-metadata-field'
+);
+const strategyResult: StrategyResult = {
+  resultCount: 1,
+  products: [
+    {
+      id: 'test-product-id',
+      name: 'test-product',
+      description: 'test-product',
+      brand: 'test-brand',
+      pageUrl: 'http://some-product-url',
+      thumbNailImage: 'http://some-thumbnail-imgae-url',
+      mainImage: 'http://some-main-imgae-url',
+      price: 20.99,
+      metadata: productMetadata,
+    },
+  ],
+  paged: {
+    from: 1,
+    size: 5,
+  },
+  metadata: strategyResultMetadata,
+};
+
+class MockCdsEndpointsService {
+  getUrl = createSpy('StrategyAdapter.loadProductsForStrategy').and.callFake(
+    endpoint => endpoint
+  );
+}
+
+describe('MerchandisingStrategyAdapter', () => {
+  let strategyAdapter: MerchandisingStrategyAdapter;
+  let httpMock: HttpTestingController;
+
+  beforeEach(() => {
+    TestBed.configureTestingModule({
+      imports: [HttpClientTestingModule],
+      providers: [
+        {
+          provide: CdsEndpointsService,
+          useClass: MockCdsEndpointsService,
+        },
+        MerchandisingStrategyAdapter,
+      ],
+    });
+
+    httpMock = TestBed.get(HttpTestingController as Type<
+      HttpTestingController
+    >);
+    strategyAdapter = TestBed.get(MerchandisingStrategyAdapter as Type<
+      MerchandisingStrategyAdapter
+    >);
+  });
+
+  afterEach(() => {
+    httpMock.verify();
+  });
+
+  it('should be created', () => {
+    expect(strategyAdapter).toBeTruthy();
+  });
+
+  describe('load products for strategy', () => {
+    it('should load products for the given strategy id', () => {
+      strategyAdapter.loadProductsForStrategy(strategyId).subscribe(result => {
```

Comment:
The same comment for subscription as in `projects/cds/src/merchandising/connectors/strategy/strategy.connector.spec.ts`

Reply:
As `unsubscribe` closes resources we found that the http request was never being made to the mock client if we did it that way. We have updated all other tests to use the `unsubscribe` method, but it looks like even in the occ adapter tests that make an http call it asserts in the subscribe. If there is a new way of doing it where the expect is not in the subscribe we'll happily update this test

Reply:
Yes, it's actually ok to have it like this in this case, sorry about the noise.